File: mobile/ios/WaldoKit/Sources/WaldoKit/DesignSystem/Tokens/ColorTokens.swift
import SwiftUI

extension Color {
    /// `0xRRGGBB` convenience initializer for the fixed design-token defaults below.
    init(hex: UInt32) {
        let r = Double((hex >> 16) & 0xFF) / 255
        let g = Double((hex >> 8) & 0xFF) / 255
        let b = Double(hex & 0xFF) / 255
        self.init(red: r, green: g, blue: b)
    }
}

/// specs/004-ios-client.md §2.1 — the semantic color vocabulary, identical to the Android client's
/// token names. Components read these fields ONLY, never a literal `Color(...)`.
public struct ColorTokens: Equatable {
    public var primary: Color
    public var onPrimary: Color
    public var secondary: Color
    public var surface: Color
    public var onSurface: Color
    public var surfaceVariant: Color
    public var danger: Color
    public var onDanger: Color
    public var success: Color
    public var warning: Color
    public var outline: Color

    public init(
        primary: Color, onPrimary: Color, secondary: Color, surface: Color, onSurface: Color,
        surfaceVariant: Color, danger: Color, onDanger: Color, success: Color, warning: Color, outline: Color
    ) {
        self.primary = primary
        self.onPrimary = onPrimary
        self.secondary = secondary
        self.surface = surface
        self.onSurface = onSurface
        self.surfaceVariant = surfaceVariant
        self.danger = danger
        self.onDanger = onDanger
        self.success = success
        self.warning = warning
        self.outline = outline
    }

    // "Waldo — Family Location Design System" (design/waldo-design-system/, 2026-07-20).
    // Calm teal-forward palette; every text/essential-icon pairing verified WCAG 2.1 AA
    // (ratios in design/waldo-design-system/README.md and specs/004 §2.1).
    public static let light = ColorTokens(
        primary: Color(hex: 0x00696E), onPrimary: Color(hex: 0xFFFFFF), secondary: Color(hex: 0x4C5FD5),
        surface: Color(hex: 0xFAFAF7), onSurface: Color(hex: 0x1B1D1C), surfaceVariant: Color(hex: 0xEEEEE9),
        danger: Color(hex: 0xC0362C), onDanger: Color(hex: 0xFFFFFF), success: Color(hex: 0x1E7D46),
        warning: Color(hex: 0x8A5A00), outline: Color(hex: 0xC9C8C2)
    )

    public static let dark = ColorTokens(
        primary: Color(hex: 0x4CD4D9), onPrimary: Color(hex: 0x00312F), secondary: Color(hex: 0xA9B4FF),
        surface: Color(hex: 0x17181A), onSurface: Color(hex: 0xECECE6), surfaceVariant: Color(hex: 0x24262A),
        danger: Color(hex: 0xF2867B), onDanger: Color(hex: 0x490A05), success: Color(hex: 0x5FD08A),
        warning: Color(hex: 0xE4B44C), outline: Color(hex: 0x3A3D42)
    )
}

File: mobile/android/app/src/main/java/com/whereswaldo/android/ui/designsystem/token/ColorTokens.kt
package com.whereswaldo.android.ui.designsystem.token

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color

/**
 * The color half of the design-token contract (specs/003-android-client.md §4.1/§4.2).
 * These are the ONLY color values allowed to exist anywhere in the app outside this file —
 * every component reads [com.whereswaldo.android.ui.designsystem.WaldoTheme.colors] instead of
 * constructing a [Color] literal.
 */
@Immutable
data class WaldoColorTokens(
    val primary: Color,
    val onPrimary: Color,
    val secondary: Color,
    val surface: Color,
    val onSurface: Color,
    val surfaceVariant: Color,
    val danger: Color,
    val onDanger: Color,
    val success: Color,
    val warning: Color,
    val outline: Color,
)

/**
 * "Waldo — Family Location Design System" (design/waldo-design-system/, 2026-07-20).
 * Calm teal-forward palette; every text/essential-icon pairing verified WCAG 2.1 AA
 * (ratios in design/waldo-design-system/README.md and specs/003 §4.2). Fully swappable via this seam.
 */
val LightWaldoColors = WaldoColorTokens(
    primary = Color(0xFF00696E),
    onPrimary = Color(0xFFFFFFFF),
    secondary = Color(0xFF4C5FD5),
    surface = Color(0xFFFAFAF7),
    onSurface = Color(0xFF1B1D1C),
    surfaceVariant = Color(0xFFEEEEE9),
    danger = Color(0xFFC0362C),
    onDanger = Color(0xFFFFFFFF),
    success = Color(0xFF1E7D46),
    warning = Color(0xFF8A5A00),
    outline = Color(0xFFC9C8C2),
)

val DarkWaldoColors = WaldoColorTokens(
    primary = Color(0xFF4CD4D9),
    onPrimary = Color(0xFF00312F),
    secondary = Color(0xFFA9B4FF),
    surface = Color(0xFF17181A),
    onSurface = Color(0xFFECECE6),
    surfaceVariant = Color(0xFF24262A),
    danger = Color(0xFFF2867B),
    onDanger = Color(0xFF490A05),
    success = Color(0xFF5FD08A),
    warning = Color(0xFFE4B44C),
    outline = Color(0xFF3A3D42),
)

val LocalWaldoColors = staticCompositionLocalOf { LightWaldoColors }
